ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeStyle/FantomasOptionsPage.fs

```diff
@@ -0,0 +1,350 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
+
+open System
+open System.Collections.Generic
+open System.IO
+open System.Runtime.InteropServices
+open JetBrains.Application.Notifications
+open JetBrains.Application.UI.Components
+open JetBrains.Application.UI.Options
+open JetBrains.Application.UI.Options.OptionsDialog
+open JetBrains.Collections.Viewable
+open JetBrains.ProjectModel
+open JetBrains.ProjectModel.NuGet.DotNetTools
+open JetBrains.ReSharper.Plugins.FSharp.Settings
+open JetBrains.ReSharper.Resources.Resources.Icons
+open JetBrains.IDE.UI.Extensions
+open JetBrains.Rider.Model
+open JetBrains.Rider.Model.UIAutomation
+open JetBrains.Threading
+open JetBrains.IDE.UI.Extensions.Validation
+
+[<AutoOpen>]
+module private FantomasVersions =
+    let [<Literal>] MinimalSupportedVersion = "3.2"
+    let [<Literal>] BundledVersion = "4.5.11"
+
+type FantomasValidationResult =
+    | Ok
+    | FailedToRun
+    | UnsupportedVersion
+    | SelectedButNotFound
+
+type FantomasVersion =
+    | Bundled
+    | SolutionDotnetTool
+    | GlobalDotnetTool
+
+type FantomasRunSettings =
+    { Version: FantomasVersion * string
```

Comment:
The purpose of the `string` item is very hard to guess. Could you redesign this part, so it's easier to understand, please? Could `Version` type better fit the version role?

Reply:
We often want to have a string representation of the version and use `Version` just to validate it. Perhaps a `Version option` would be better since the found version can be invalid.